blocks/main_loop.py

```diff
@@ -42,6 +45,87 @@
 a model. Check documentation of your extensions."""
 
 
+class Timer(object):
+    def __init__(self):
+        self.total = defaultdict(int)
+        self.current = []
+        self.order = OrderedSet()
+
+    def enter(self, name):
+        self.current.append(name)
+        # We record the order in which sections were first called
+        self.order.add(tuple(self.current))
+
+    def exit(self, t):
+        self.total[tuple(self.current)] += t
+        self.current.pop()
+
+    def report(self):
+        """Print a report of timing information to standard output.
+
+        .. todo::
+
+           This method could accept different I/O objects (e.g. standard
```

Comment:
And then I load my main loop pickle in an IPython notebook and want to print report to standard output.

I think it's basic functionality and requires roughly 3 lines to support.